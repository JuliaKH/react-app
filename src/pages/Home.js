import React, {Component} from 'react';
import './ImageList/ImageList.scss'
import {Link} from "react-router-dom";
import PropTypes from "prop-types";

class ImageList extends Component {

    componentDidMount() {
        this.props.userSubmit('popular');
    }

    render() {
        this.imgs = this.props.foundImages.map(img => {
            return <Link to={img.id} key={img.id}> <img  src={img.urls.regular} alt={img.alt_description} className="images__list-img"/></Link>
        });
        return (
            <section className="images">
                <div className="container">
                    <div className="images__list">
                        {this.imgs}
                    </div>
                </div>
            </section>
        );
    }
}
ImageList.propTypes = {
    foundImages: PropTypes.array,
    userSubmit:PropTypes.func
};
export default ImageList;
