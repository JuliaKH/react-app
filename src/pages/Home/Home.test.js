import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Home from "./Home";

describe('Home ', () => {

    const mockUserSubmit = jest.fn(() => 'Set new query');
    const mockFoundImages = [
        {
            id: '123',
            urls:{
                regular: 'some.jpg'
            },
            alt_description: 'alt'
        }
    ];

    it('should render correctly', () => {
        const wrapper = shallow(<Home foundImages={mockFoundImages} userSubmit={mockUserSubmit}/>);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

});
