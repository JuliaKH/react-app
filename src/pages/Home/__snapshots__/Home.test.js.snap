// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home  should render correctly 1`] = `
<section
  className="images"
>
  <div
    className="container"
  >
    <div
      className="images__list"
    >
      <Link
        key="123"
        to="/images/123"
      >
         
        <img
          alt="alt"
          className="images__list-img"
          src="some.jpg"
        />
      </Link>
    </div>
  </div>
</section>
`;
